lib/puma/cluster.rb

```diff
@@ -37,7 +37,14 @@ def stop_workers
       @workers.each { |x| x.term }
 
       begin
-        @workers.each { |w| Process.waitpid(w.pid, Process::WNOHANG) }
+        pids = @workers.map(&:pid)
+        until pids.empty?
+          if p = Process.waitpid(-1, Process::WNOHANG)
+            pids.delete(p)
```

Comment:
`p` is reserved (`Kernel#p`), it's even highlights so, and [it's smelling](https://github.com/troessner/reek/blob/8a2bd23/docs/Uncommunicative-Variable-Name.md). What about something like `process`?

Reply:
Thanks! Fixed.

Reply:
Good point.  My mistake